challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map(&:capitalize)
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |i| i == 'dog' }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |i| i == 'dog' }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select.grep(/^[cd]/)
```

Comment:
You shouldn't need to call `#select` first, just the grep call on the collection will work :)